packages/arb-token-bridge-ui/src/components/common/WithdrawalCountdown.tsx

```diff
@@ -2,47 +2,73 @@ import { useMedia } from 'react-use'
 import dayjs, { Dayjs } from 'dayjs'
 
 import { useNetworksAndSigners } from '../../hooks/useNetworksAndSigners'
-import { ChainId } from '../../util/networks'
+import {
+  getBaseChainIdByChainId,
+  getBlockTime,
+  getConfirmPeriodBlocks
+} from '../../util/networks'
 
 /**
- * Buffer for after a node is confirmable but isn't yet confirmed; we give 30 minutes, should usually/always be less in practice.
+ * Buffer for after a node is confirmable but isn't yet confirmed.
+ * A rollup block (RBlock) typically gets asserted every 30-60 minutes.
  */
-const CONFIRMATION_BUFFER_MINUTES = 30
+const CONFIRMATION_BUFFER_MINUTES = 60
 
 function getTxConfirmationDate({
   createdAt,
-  parentChainId
+  withdrawalFromChainId,
+  baseChainId
 }: {
   createdAt: Dayjs
-  parentChainId: number
+  withdrawalFromChainId: number
+  baseChainId: number
 }) {
-  const confirmNodeMinutes = getNodeConfirmationTimeInMinutes(parentChainId)
+  const { confirmationMins } = getTxConfirmationTime({
+    withdrawalFromChainId,
+    baseChainId
+  })
 
-  return createdAt
-    .add(confirmNodeMinutes, 'minute')
-    .add(CONFIRMATION_BUFFER_MINUTES, 'minute')
+  return createdAt.add(confirmationMins, 'minute')
 }
 
-function getTxConfirmationRemainingMinutes({
-  createdAt,
-  parentChainId
-}: {
-  createdAt: Dayjs
-  parentChainId: number
-}) {
-  const txConfirmationDate = getTxConfirmationDate({ createdAt, parentChainId })
+function getTxConfirmationRemainingMinutes(txConfirmationDate: dayjs.Dayjs) {
```

Comment:
Repeating the comment since it's valid here as well for all functions - we should use methods provided by dayjs to offload date-time calculations and formatting, and then we won't be needing confirmationSeconds, confirmationMins, confirmationHours, confirmationDays.

https://day.js.org/docs/en/display/from-now

Reply:
good point! updated